src/script/lua_api/l_mapgen.cpp

```diff
@@ -1322,14 +1322,15 @@ int ModApiMapgen::l_register_ore(lua_State *L)
 	BiomeManager *bmgr    = emerge->getWritableBiomeManager();
 	OreManager *oremgr    = emerge->getWritableOreManager();
 
-	enum OreType oretype = (OreType)getenumfield(L, index,
-				"ore_type", es_OreType, ORE_SCATTER);
-	Ore *ore = oremgr->create(oretype);
-	if (!ore) {
-		errorstream << "register_ore: ore_type " << oretype << " not implemented\n";
+	int oretype_int;
+	std::string oretype_string = getstringfield_default(L, index, "ore_type", "nil");
+	if (!string_to_enum(es_OreType, oretype_int, oretype_string)) {
+		errorstream << "register_ore: unknown oretype \"" << oretype_string << "\"" << std::endl;
 		return 0;
 	}
+	enum OreType oretype = (OreType) oretype_int;
 
+	Ore *ore = oremgr->create(oretype);
```

Comment:
Optional: this could be changed to a `std::unique_ptr` to ensure RAII. Release and forward the pointer at the function end to make most use of this feature.

Reply:
I modernized it a little bit now, but couldn't release the pointer at the "right" place, because it is still needed after the transfer to the `oremgr` and I didn't want to make a copy.